src/builder/QuarkBuilderBase.sol

```diff
@@ -0,0 +1,642 @@
+                        List.addAction(actions, bridgeActions[j]);
+                    }
+                } else {
+                    uint256 balanceOnChain = getBalanceOnChain(
+                        actionIntent.assetSymbolOuts[i], actionIntent.chainId, chainAccountsList, payment
+                    );
+                    uint256 amountNeededOnChain = getAmountNeededOnChain(
+                        actionIntent.assetSymbolOuts[i], actionIntent.amountOuts[i], actionIntent.chainId, payment
+                    );
+                    uint256 maxCostOnChain =
+                        payment.isToken ? PaymentInfo.findMaxCost(payment, actionIntent.chainId) : 0;
+                    uint256 availableAssetBalance =
+                        balanceOnChain >= maxCostOnChain ? balanceOnChain - maxCostOnChain : 0;
+                    revert FundsUnavailable(actionIntent.assetSymbolOuts[i], amountNeededOnChain, availableAssetBalance);
+                }
+            }
+        }
+
+        if (payment.isToken && !paymentTokenInAssetSymbolOut) {
+            uint256 maxCostOnDstChain = PaymentInfo.findMaxCost(payment, actionIntent.chainId);
+
+            for (uint256 k = 0; k < actionIntent.assetSymbolIns.length; ++k) {
+                if (Strings.stringEqIgnoreCase(actionIntent.assetSymbolIns[k], payment.currency)) {
+                    maxCostOnDstChain = Math.subtractFlooredAtZero(maxCostOnDstChain, actionIntent.amountIns[k]);
+                }
+            }
+
+            if (
+                needsBridgedFunds(payment.currency, maxCostOnDstChain, actionIntent.chainId, chainAccountsList, payment)
+            ) {
+                if (actionIntent.bridgeEnabled) {
+                    (IQuarkWallet.QuarkOperation[] memory bridgeQuarkOperations, Actions.Action[] memory bridgeActions)
+                    = Actions.constructBridgeOperations(
+                        Actions.BridgeOperationInfo({
+                            assetSymbol: payment.currency,
+                            amountNeededOnDst: maxCostOnDstChain,
+                            dstChainId: actionIntent.chainId,
+                            recipient: actionIntent.actor,
+                            blockTimestamp: actionIntent.blockTimestamp,
+                            useQuotecall: actionIntent.useQuotecall
+                        }),
+                        chainAccountsList,
+                        payment
+                    );
+
+                    for (uint256 i = 0; i < bridgeQuarkOperations.length; ++i) {
+                        List.addQuarkOperation(quarkOperations, bridgeQuarkOperations[i]);
+                        List.addAction(actions, bridgeActions[i]);
+                    }
+                } else {
+                    if (
+                        needsBridgedFunds(
+                            payment.currency, maxCostOnDstChain, actionIntent.chainId, chainAccountsList, payment
+                        )
```

Comment:
This if statement is redundant.

Reply:
oops forgot to trim it during copy-pasta haha